src/app/core/services/navigation/nav-tree.service.ts

```diff
@@ -38,36 +38,46 @@ export abstract class NavTreeService<ContentT extends RoutedContentInfo> {
     ),
     this.content$.pipe(this.childrenNavData())
   );
-  state$ = combineLatest([ this.children$, this.content$ ]).pipe(
-    mergeScan((prevState: FetchState<NavTreeData>, [ children, content ]) => {
-
+  state$ = combineLatest([
+    this.children$,
+    this.content$.pipe(switchMap(content => this.currentContent.contentHasChanged$.pipe(
+      map(() => true),
+      startWith(false),
+      map(hasChanged => ({ content, hasChanged }))
+    ))),
+  ]).pipe(
+    mergeScan((prevState: FetchState<NavTreeData>, [ children, { content, hasChanged }]) => {
```

Comment:
If there is a content with "hasChanged=true" and then `children` emit, the next mergeScan block will use `hasChanged=true`and so consider it has to refresh while it should not. Isn't it?

Reply:
Fixed
Addressing it also fixed [this comment](https://github.com/France-ioi/AlgoreaFrontend/pull/957#discussion_r817749219)